### PR TITLE
Blockchain package tests and CI parallel tests

### PR DESCRIPTION
**Pull request by stuckDaemon**
## Scope

This PR includes blockchain package test coverage and fix plus a new workflow file for CI. It will run parallel tests all across the packages with a specific job for blockchain package.

## Why?

This is part of the group of tickets aiming to increase test coverage and code rubustness

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -131,23 +130,95 @@ jobs:
...
+  golang-blockchain-tests:
+    runs-on: ${{ matrix.os }}
+    needs: golang-build
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-20.04 ]
+        test-cmd: [
+          github.com/MadBase/MadNet/blockchain/dkg/dkgtasks -run TestShareDistribution_Group_1,
```
Comment (vtleonardo) [inline .github/workflows/ci.yml:161]: Are these not going to run the tests in the remote package?

File: blockchain/cancellation_test.go [hunk 2]
```diff
@@ -32,7 +32,7 @@ func TestSleepWithContextComplete(t *testing.T) {
 	assert.True(t, completed)
 }
 
-func TestSleepWithContextInterupted(t *testing.T) {
+func TestCancellation_SleepWithContextInterrupted(t *testing.T) {
```
Comment (vtleonardo) [inline blockchain/cancellation_test.go:35]: is this  `TestCancellation` a special golang test keyword(e.g to skip the test)? Or is just a rename of the tests?

File: blockchain/dkg/dkgtasks/dispute_missing_key_shares_task_test.go [hunk 3]
```diff
@@ -57,7 +58,7 @@ func TestDisputeMissingKeySharesTaskFourUnsubmittedKeyShare_DoWork_Success(t *te
 	for idx := 0; idx < n; idx++ {
 		state := dkgStates[idx]
 		disputeMissingKeyshareTask := suite.disputeMissingKeyshareTasks[idx]
-
+		time.Sleep(5 * time.Second)
```
Comment (vtleonardo) [inline blockchain/dkg/dkgtasks/dispute_missing_key_shares_task_test.go]: Do we need this sleep here?

File: blockchain/dkg/dkgtasks/mpk_submission_task_test.go [hunk 4]
```diff
@@ -49,6 +49,7 @@ func TestMPKSubmissionGoodAllValid(t *testing.T) {
 
 		}
 	}
+	time.Sleep(5 * time.Second)
```
Comment (vtleonardo) [inline blockchain/dkg/dkgtasks/mpk_submission_task_test.go]: Do we need this sleep here?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (stuckDaemon) [reply .github/workflows/ci.yml:161]: Which one is the `remote` package?

File: blockchain/cancellation_test.go (hunk 2 above)
Comment (stuckDaemon) [reply blockchain/cancellation_test.go:35]: Just renaming.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (vtleonardo) [reply .github/workflows/ci.yml:161]: Running using the github package it's not going to download the package from github master and run from there? Or is it going to grab the local files?

Comment (stuckDaemon) [reply .github/workflows/ci.yml:161]: I guess it's running the local files as in this case `TestShareDistribution_Group_1` won't exist on `master` branch.

**Pull request merged**